dev/CommandBarFlyout/CommandBarFlyout.cpp

```diff
@@ -14,32 +14,31 @@
 bool CommandBarFlyoutTrace::s_IsDebugOutputEnabled{ false };
 bool CommandBarFlyoutTrace::s_IsVerboseDebugOutputEnabled{ false };
 
-// List of AppBarButton dependency properties being listened to for raising the CommandBarFlyoutCommandBar::OnCommandBarElementDependencyPropertyChanged notifications.
-// AppBarButton::IsCompact and AppBarButton::LabelPosition having no effect on an AppBarButton's rendering, when used as a secondary command, they are not present in the list.
-winrt::DependencyProperty CommandBarFlyout::s_appBarButtonDependencyProperties[s_commandBarElementDependencyPropertiesCount]
-{
-    winrt::AppBarButton::IconProperty(),
-    winrt::AppBarButton::LabelProperty(),
-    nullptr
-};
-
-// List of AppBarToggleButton dependency properties being listened to for raising the CommandBarFlyoutCommandBar::OnCommandBarElementDependencyPropertyChanged notifications.
-// AppBarToggleButton::IsCompact and AppBarToggleButton::LabelPosition having no effect on an AppBarToggleButton's rendering, when used as a secondary command, they are not present in the list.
-winrt::DependencyProperty CommandBarFlyout::s_appBarToggleButtonDependencyProperties[s_commandBarElementDependencyPropertiesCount]
-{
-    winrt::AppBarToggleButton::IconProperty(),
-    winrt::AppBarToggleButton::LabelProperty(),
-    nullptr
-};
+// List of AppBarButton/AppBarToggleButton dependency properties being listened to for raising the CommandBarFlyoutCommandBar::OnCommandBarElementDependencyPropertyChanged notifications.
+// IsCompact and LabelPosition have no effect on an AppBarButton's rendering, when used as a secondary command, they are not present in the list.
+// These two arrays are initialized in the constructor instead of being statically initialized here because that would result in the initialization happening during
+// dllmain and it is not valid to call COM apis at that time.
+winrt::DependencyProperty CommandBarFlyout::s_appBarButtonDependencyProperties[s_commandBarElementDependencyPropertiesCount]{ nullptr, nullptr, nullptr };
```

Comment:
>{ nullptr, nullptr, nullptr }; [](http://example.com/codeflow?start=124&length=30)

I don't love that this is required to initialize the array to nullptrs.
It is not possible to just use "{}", because the default construction of a winrt::Foo object is not null - it creates a new instance of the object. Or at least it does if the winrt type itself has a default constructor. DependencyProperty does not have a constructor, since they are never instantiated directly (instead they are returned by DependencyProperty::Register). So in this case "{}" would be a compiler error.